File: javascript/MongoCursorTest.js

var tests = [


    function testThatThe10thScifiMovieOrderedByRatingIsTheExpectedOne() {
        query = {"genres" : "Sci-Fi"};
        cursor = db.movies.find(query).skip(9).limit(1).sort({"rating":1});
        assert(cursor.count(), 1);
        var movie = cursor.next();
        assert.eq(movie["title"], 'Space Mutiny');

     }
]

for (var i=0 ; i < tests.length ; i++) {
	tests[i]();
}

File: python/MongoCursorTest.py
# -*- coding: utf-8 -*-
import unittest
import pymongo
from pymongo import MongoClient


class MongoCursorTest(unittest.TestCase):

    #
    # Initialisation des membres privés à compléter
    def setUp(self):
        self.mongoclient = MongoClient()
        self.db = self.mongoclient.mediatheque
        self.movies = self.db.movies


    def tearDown(self): 
        self.mongoclient.close()


    # Ce test permet de vérifier le titre du 10eme film trié par note ascendante
    def testThatThe10thScifiMovieOrderedByRatingIsTheExpectedOne(self):
      cursor = self.movies.find({"genres" : "Sci-Fi"}).skip(9).limit(1).sort([("rating",1)])
      for movie in cursor:
        self.assertEqual(movie["title"], 'Space Mutiny')


if __name__ == '__main__':
    unittest.main()
